test/units/module_utils/facts/hardware/linux/test_get_pool.py

```diff
@@ -17,7 +17,7 @@ def test_get_pool():
 
 def test_get_dummy_pool():
 
-    with mock.patch('multiprocessing.pool.ThreadPool', new=mock.MagicMock()) as TPool:
-        TPool.__new__ = mock.MagicMock(side_effect=PermissionError('pooof'))
+    with mock.patch('multiprocessing.pool.ThreadPool.__new__', new=mock.MagicMock()) as TPool:
+        TPool = mock.MagicMock(side_effect=PermissionError('pooof'))
```

Comment:
The mock can simplified further:

```suggestion
    with mock.patch.object(multiprocessing.pool.ThreadPool, '__init__', side_effect=PermissionError('poof')):
```